File: buildSrc/src/main/java/com/agfa/typeddicom/gradleplugin/ProcessDicomXmlPluginExtension.java
package com.agfa.typeddicom.gradleplugin;

import org.gradle.api.file.DirectoryProperty;
import org.gradle.api.tasks.InputDirectory;
import org.gradle.api.tasks.OutputDirectory;

public interface ProcessDicomXmlPluginExtension {
    @InputDirectory
    DirectoryProperty getDicomStandardXmlDirectory();

    @InputDirectory
    DirectoryProperty getMustacheTemplateDirectory();

    @OutputDirectory
    DirectoryProperty getGeneratedJavaOutputDirectory();
}

File: buildSrc/src/main/kotlin/com/agfa/typeddicom/ProcessDicomXmlPluginExtension.kt
package com.agfa.typeddicom;

import org.gradle.api.file.DirectoryProperty
import org.gradle.api.tasks.InputDirectory
import org.gradle.api.tasks.OutputDirectory

abstract class ProcessDicomXmlPluginExtension {
    @get:OutputDirectory
    abstract val dicomStandardXmlDirectory: DirectoryProperty

    @get:InputDirectory
    abstract val mustacheTemplateDirectory: DirectoryProperty

    @get:InputDirectory
    abstract val generatedJavaOutputDirectory: DirectoryProperty
}
